### DISCUSSION TITLE
CMake of EnvelopeProblemSOS failing to compile

### DISCUSSION THREAD

**Discussion by shahpratham** [GeomScale/volesti#209] [Q&A] [answered]
I am working on #193 
- I added the feature in `cmake-gcc.yml` to compile all CMakeLists.txt in github-actions. It compiles all CMake files inside each example folder.
- I have fixed the CMake of `vpolypote-volume`.
- The only thing left is the CMake of `EnvelopeProblemSOS`. While changing `Python2` to `Python3` in CMake, it was working and the build was successful in my system, but later when I tried to push it on my branch, I got the error `Could NOT find Python3 (missing: Python3_NumPy_INCLUDE_DIRS NumPy) (found version "3.6.9")` and `No header defined for *boost; skipping header check`.

[This](https://github.com/shahpratham/volesti/runs/5300486542?check_suite_focus=true) is the log of the error.
[This](https://github.com/shahpratham/volesti/blob/pratham/.github/workflows/cmake-gcc.yml) is my `cmake-gcc.yml` file.

Can anyone suggest what I should do to resolve this? I have installed Boost dependencies and NumPy in the `cmake-gcc.yml` file.

**Comment by vaithak** [comment, marked as answer]
Hi @shahpratham, you may have to debug/fix the CMake file for `EnveleopeProblemSOS`, some issues that I found which may or may not be relevant but worth checking out:
- The CMakeLists.txt of this example uses Python2.7 and you are downloading python3 in `cmake-gcc.yaml`, either try using python3 with the example or change `cmake-gcc.yaml`.
- Another issue being that the example is not using the `external/cmake-files/Boost.cmake` for downloading Boost but instead using a different method which may require some extra downloading steps for running it on CircleCI.

**Comment by shahpratham** [comment]
Hey @vaithak, thank you for replying.

> either try using python3 with the example

yes, I was using python3 with CmakeLists, but I later found numpy was not being installed in github actions, so i added `"actions/setup-python@v2"` in github actions and now its working.

> Another issue being that the example is not using the external/cmake-files/Boost.cmake for downloading Boost

I included Boost.cmake from this path and its working.

Now, this example is successfully compiling in gtihub actions.  

Thanks.
